File: serveur/app/models/exercise_muscle.py
# Exemple : app/models/exercise_muscle.py

from sqlalchemy import Column, Integer, Float, ForeignKey, String, DateTime, func
from app.database import Base

import uuid

class ExerciseMuscle(Base):
    __tablename__ = "exercise_muscles"

    id = Column(Integer, primary_key=True, index=True)

    exercise_uuid = Column(String, ForeignKey("exercises.uuid", ondelete="CASCADE"), nullable=False, index=True)
    muscle_uuid = Column(String, ForeignKey("muscles.uuid", ondelete="CASCADE"), nullable=False, index=True)
    coefficient = Column(Float, nullable=False, default=1.0, server_default="1")

    updated_at = Column(DateTime(timezone=True), nullable=True)

    uuid = Column(String, nullable=False, default=lambda: str(uuid.uuid4()), unique=True)

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/ExerciseMuscle.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.ColumnInfo
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "exercise_muscles",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)],
    foreignKeys = [
        androidx.room.ForeignKey(
            entity = Exercise::class,
            parentColumns = ["uuid"],
            childColumns = ["exercise_uuid"],
            onDelete = androidx.room.ForeignKey.CASCADE
        ),
        androidx.room.ForeignKey(
            entity = Muscle::class,
            parentColumns = ["uuid"],
            childColumns = ["muscle_uuid"],
            onDelete = androidx.room.ForeignKey.CASCADE
        )
    ],
)
data class ExerciseMuscle(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "exercise_uuid") val exerciseUUID: String,
    @ColumnInfo(name = "muscle_uuid") val muscleUUID: String,
    val coefficient: Float = 1.0f,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
